libdevice/fallback-cassert.cpp

```diff
@@ -7,99 +7,8 @@
 //===----------------------------------------------------------------------===//
 
 #include "atomic.hpp"
-#include "include/assert-happened.hpp"
 #include "wrapper.h"
 
-#if defined(__SPIR__) || defined(__SPIRV__)
-
-#define ASSERT_NONE 0
-#define ASSERT_START 1
-#define ASSERT_FINISH 2
-
-// definition
-SPIR_GLOBAL AssertHappened SPIR_AssertHappenedMem;
-
-DEVICE_EXTERN_C void __devicelib_assert_read(void *_Dst) {
-  AssertHappened *Dst = (AssertHappened *)_Dst;
-  int Flag = atomicLoad(&SPIR_AssertHappenedMem.Flag);
-
-  if (ASSERT_NONE == Flag) {
-    Dst->Flag = Flag;
-    return;
-  }
-
-  if (Flag != ASSERT_FINISH)
-    while (ASSERT_START == atomicLoad(&SPIR_AssertHappenedMem.Flag))
-      ;
-
-  *Dst = SPIR_AssertHappenedMem;
-}
-
-DEVICE_EXTERN_C void __devicelib_assert_fail(const char *expr, const char *file,
```

Comment:
Looking at the failures, we may still need a dummy implementation of this. The extension [states](https://github.com/intel/llvm/blob/sycl/sycl/doc/extensions/supported/sycl_ext_oneapi_assert.asciidoc#extension-to-enum-class-aspect):

> If device has the `ext_oneapi_native_assert` aspect, then its Device-Side Runtime is capable of native support of assert. If device doesn’t have the aspect, then assertions on the device will be silently ignored.